### ISSUE TITLE
Redirect in workflow prevents creation of new users

### ISSUE THREAD

**Issue by JohnStov** [OrchardCMS/Orchard#7690]
I have a workflow which runs on the log in event. It redirects to a dynamic form to gather further information from the user under certain circumstances. If it runs immediately after a user has created a new account, the user account is never created and no UserPart is available when the Dynamic Form data is posted back.

**Comment by JohnStov** [comment]
This problem only occurs when a user creates a new account through the registration page. The UserLoggedIn workflow runs in the same request, so when the redirect occurs, the response is forcibly terminated and the UserPart is never flushed - the exception causes NHibernate to rollback the transaction.

This means that when we redirect, although we have a cookie for the user, there is no corresponding entry in the users table.

I thought we could fix it by calling HttpResponse.Redirect(url, false). This will allow the response to finish, so NHibernate can flush the UserPart to the db. 

Unfortunately,Orchard itself seems to call HttpResponse.Redirect, later in the request, to send the user to the homepage. This overrides the redirect that we tried to set up in the workflow.

We need to either 1) force the DB transaction to close so that the user is created properly before we redirect, or 2) tell Orchard that when the request is done, not to redirect to the home page, but to the url we specified in the workflow.

Any suggestions as to which approach is preferable, and hints as to how to do it gratefully received :)
